observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemAdd.java

```diff
@@ -28,42 +28,41 @@
+import org.wildfly.extension.opentelemetry.api.WildFlyOpenTelemetryConfig;
 
 /**
  * Handler responsible for adding the subsystem resource to the model
  *
  * @author <a href="jasondlee@redhat.com">Jason Lee</a>
  */
-public class OpenTelemetrySubsystemAdd extends AbstractBoottimeAddStepHandler {
+class OpenTelemetrySubsystemAdd extends AbstractBoottimeAddStepHandler {
 
-    private OpenTelemetrySubsystemAdd() {
+    OpenTelemetrySubsystemAdd() {
         super(OpenTelemetrySubsystemDefinition.ATTRIBUTES);
     }
 
-    public static final OpenTelemetrySubsystemAdd INSTANCE = new OpenTelemetrySubsystemAdd();
-
     /**
      * {@inheritDoc}
      */
     @Override
     protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
         super.performBoottime(context, operation, model);
 
-        final OpenTelemetryConfig config = OpenTelemetryConfigBuilder.config()
-                .withServiceName(OpenTelemetrySubsystemDefinition.SERVICE_NAME.resolveModelAttribute(context, model).asStringOrNull())
-                .withExporter(OpenTelemetrySubsystemDefinition.EXPORTER.resolveModelAttribute(context, model).asString())
-                .withEndpoint(OpenTelemetrySubsystemDefinition.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull())
-                .withSpanProcessor(OpenTelemetrySubsystemDefinition.SPAN_PROCESSOR_TYPE.resolveModelAttribute(context, model).asString())
-                .withBatchDelay(OpenTelemetrySubsystemDefinition.BATCH_DELAY.resolveModelAttribute(context, model).asLong())
-                .withMaxQueueSize(OpenTelemetrySubsystemDefinition.MAX_QUEUE_SIZE.resolveModelAttribute(context, model).asInt())
-                .withMaxExportBatchSize(OpenTelemetrySubsystemDefinition.MAX_EXPORT_BATCH_SIZE.resolveModelAttribute(context, model).asInt())
-                .withExportTimeout(OpenTelemetrySubsystemDefinition.EXPORT_TIMEOUT.resolveModelAttribute(context, model).asLong())
-                .withSampler(OpenTelemetrySubsystemDefinition.SAMPLER.resolveModelAttribute(context, model).asStringOrNull())
-                .withRatio(OpenTelemetrySubsystemDefinition.RATIO.resolveModelAttribute(context, model).asDoubleOrNull())
-                .build();
-        final OpenTelemetryHolder holder = new OpenTelemetryHolder(config);
+        final WildFlyOpenTelemetryConfig config = new WildFlyOpenTelemetryConfig(
+                OpenTelemetrySubsystemDefinition.SERVICE_NAME.resolveModelAttribute(context, model).asStringOrNull(),
+                OpenTelemetrySubsystemDefinition.EXPORTER.resolveModelAttribute(context, model).asString(),
+                OpenTelemetrySubsystemDefinition.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull(),
+                OpenTelemetrySubsystemDefinition.BATCH_DELAY.resolveModelAttribute(context, model).asStringOrNull(),
```

Comment:
From "long" to "String" for a parameter: This may cause errors at a later point of runtime, correct? Are you forced to degrade the type? (Just as a sample here, there are some more)

Reply:
I did that largely because the otel config map has to be String,String. I've reverted this, though, and moved the string conversion deeper into the module. Here, then, as WF reads the config, we'll get some type safety at the model level.